### ISSUE TITLE
`I001` should not be triggered for one new line after imports

### ISSUE THREAD

**Issue by samuelcolvin** [astral-sh/ruff#14836]
Background: see https://github.com/pydantic/pytest-examples/issues/46, in [pytest-examples](https://github.com/pydantic/pytest-examples) we want to allow one new line between functions to keep examples short.

This works well in general, but breaks when you have a function directly after imports where IMHO ruff `I001` is overreaching in complaining about "un-sorted or un-formatted" because there's one blank line, not two.

(We use black for formatting as it's faster than ruff running as a separate subprocess (side note, would be great if you could release `ruff` as a Python library so we can use it embeded), and ruff for linting.)

I don't think the error `Import block is un-sorted or un-formatted` makes sense for this code:

```py
from typing import Union

from pydantic_ai import RunContext, Tool
from pydantic_ai.tools import ToolDefinition

async def only_if_42(
    ctx: RunContext[int], tool_def: ToolDefinition
) -> Union[ToolDefinition, None]:
    if ctx.deps == 42:
        return tool_def

def hitchhiker(ctx: RunContext[int], answer: str) -> str:
    return f'{ctx.deps} {answer}'

hitchhiker = Tool(hitchhiker, prepare=only_if_42)
```

There should be a new `I003` code for "expected two blank line after imports", which I could then switch off.

**Comment by samuelcolvin** [comment]
`isort.lines-after-imports` is helpful, thank you.

But on balance I do think this should be a new error code too.

**Comment by MichaReiser** [comment]
> But on balance I do think this should be a new error code too.

Possibly, but I'd prefer keeping it as is for now to match the upstream isort. We should revisit (and possibly even remove) isort as a whole.

**Comment by AlexWaygood** [comment]
> I'd prefer keeping it as is for now to match the upstream isort

I'm not disputing that lots of users find it confusing that isort is implemented as a lint check rather than part of the formatter, but FWIW, I don't think the upstream isort tool has any error codes

**Comment by MichaReiser** [comment]
> 'm not disputing that lots of users find it confusing that isort is implemented as a lint check rather than part of the formatter

Totally agree and yes, the upstream isort has no error codes. But I understand that the idea is that `ISC001` matches upstream (it's how you enable isort today)

**Comment by samuelcolvin** [comment]
FWIW I suspect ruff's usage is much greater than isort's ever was, so the name `isort` is a historical curiosity that means nothing to most users.

I think this should be re-opened.
